test/new-e2e/tests/agent-metric-logs/linux-log-agent/01_file_tailing_test.go

```diff
@@ -0,0 +1,154 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package logagent
+
+import (
+	_ "embed"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	e2e "github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+)
+
+// LinuxVMFakeintakeSuite defines a test suite for the log agent interacting with a virtual machine and fake intake.
+type LinuxVMFakeintakeSuite struct {
+	e2e.Suite[e2e.FakeIntakeEnv]
+	DevMode bool
+}
+
+//go:embed log-config/log-config.yaml
+var logConfig []byte
+
+// logsExampleStackDef returns the stack definition required for the log agent test suite.
+func logsExampleStackDef() *e2e.StackDefinition[e2e.FakeIntakeEnv] {
+
+	return e2e.FakeIntakeStackDef(
+		e2e.WithAgentParams(
+			agentparams.WithLogs(),
+			agentparams.WithIntegration("custom_logs.d", string(logConfig))))
+
+}
+
+// TestE2EVMFakeintakeSuite runs the E2E test suite for the log agent with a VM and fake intake.
+func TestE2EVMFakeintakeSuite(t *testing.T) {
+	s := &LinuxVMFakeintakeSuite{}
+	_, s.DevMode = os.LookupEnv("TESTS_E2E_DEVMODE")
```

Comment:
Is this env var documented somewhere? I see it used in one other test, but nowhere in any docs.

Reply:
This is not documented, as it is not part of the framework definition. The framework has an option `DevMode`, but it must be explicitly set from code. We can consider adding a flag and/or an env variable. Note e2e variables are prefixed by `E2E_`, we can rename it later.